[Cases] Make metrics routes internal

adcoelho opened the pull request:
Fixes #162406

## Summary

- Made `getCaseMetrics` and `getCasesMetrics` internal APIs.
- There was no documentation to begin with so there was none to remove.
- The allowed/available `features` now come from the `CaseMetricsFeature` _enum_ instead of being hardcoded everywhere.
- We now **also** check for the values passed to the `features` field in case metrics requests using io-ts.
- There was already some validation before which I decided to leave. When doing `buildHandlers` the function `checkAndThrowIfInvalidFeatures` is always called. Right now, this is only used by `getCasesMetrics` and `getCaseMetrics` where we already decode the params so that validation is redundant, but if it starts being used somewhere else it will be nice to have this extra security guarantee.

### Release Notes

The get case metrics APIs are now internal.

adcoelho committed:
Initial commit.

adcoelho committed:
Create the CaseMetricsFeature enum.
Update usages EVERYWHERE.
Fixed failing jest tests.
Fixed typescript errors.

adcoelho committed:
Fix failing jest tests.

elasticmachine commented:
Pinging @elastic/response-ops-cases (Feature:Cases)

adcoelho committed:
Fix external imports of the CaseMetricsFeature.

adcoelho committed:
Fixed build error

cnasikas commented on x-pack/plugins/cases/common/api/metrics/case.ts:
```diff
@@ -76,15 +96,15 @@ export const SingleCaseMetricsRequestRt = rt.strict({
    * The metrics to retrieve.
    */
-  features: rt.array(rt.string),
+  features: rt.array(CaseMetricsFeatureFieldRt),
```
If I remember correctly the `mttr` feature is not available for the single case request.

adcoelho replied:
Thanks for this catch, I differentiate now between Single Case or Cases Metrics Features. 👍

adcoelho committed:
Moved the metrics integration tests to internal folder.
Fixed failing tests that were using the old url.
Corrected types for single case metrics features.

kibanamachine committed:
[CI] Auto-commit changed files from 'node scripts/lint_ts_projects --fix'

kibanamachine committed:
[CI] Auto-commit changed files from 'node scripts/eslint --no-cache --fix'

kibanamachine committed:
[CI] Auto-commit changed files from 'node scripts/lint_ts_projects --fix'

adcoelho committed:
Fix imports.

adcoelho committed:
Updated expected error msg in integration test.

adcoelho committed:
Merge remote-tracking branch 'upstream/main' into make-metrics-internals

adcoelho committed:
Merge branch 'main' into make-metrics-internals

adcoelho committed:
Merge branch 'main' into make-metrics-internals

cnasikas commented on x-pack/plugins/cases/common/api/metrics/case.test.ts:
Could you please add some tests to validate for invalid features?

cnasikas reviewed (approved):
🚀

adcoelho committed:
Added schema tests for invalid feature values.

adcoelho committed:
Merge branch 'main' into make-metrics-internals

adcoelho committed:
Merge branch 'main' into make-metrics-internals

adcoelho committed:
Merge branch 'main' into make-metrics-internals

adcoelho committed:
Merge branch 'main' into make-metrics-internals

adcoelho committed:
Merge branch 'main' into make-metrics-internals

machadoum reviewed (approved):
Threat hunting explore team changes LGTM!

adcoelho committed:
Merge branch 'main' into make-metrics-internals
